### PR TITLE
[FLINK-1811] Allow passing custom buffer sizes and timeouts to Persisten...

### PR DESCRIPTION
**Pull request by rmetzger**
...tKafkaSource

### CODE REVIEW

File: flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/iterator/KafkaMultiplePartitionsIterator.java [hunk 1]
```diff
@@ -33,7 +33,9 @@
 	protected List<KafkaSinglePartitionIterator> partitions;
 	protected final int waitOnEmptyFetch;
 
-	public KafkaMultiplePartitionsIterator(String hostName, String topic, Map<Integer, KafkaOffset> partitionsWithOffset, int waitOnEmptyFetch) {
+	public KafkaMultiplePartitionsIterator(String hostName, String topic,
+										   Map<Integer, KafkaOffset> partitionsWithOffset,
```
Comment (uce) [inline flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/iterator/KafkaMultiplePartitionsIterator.java]: what happend here with the indentation? :smiley:

Comment (rmetzger) [inline flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/iterator/KafkaMultiplePartitionsIterator.java]: AFAIK its a github bug.

File: flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/PersistentKafkaSource.java [hunk 2]
```diff
@@ -211,6 +215,16 @@ public void run(Collector<OUT> collector) throws Exception {
 		}
 	}
 
+	public void setConnectTimeoutMs(int connectTimeoutMs) {
+		Preconditions.checkArgument(connectTimeoutMs > 0, "The timeout must be positive");
+		this.connectTimeoutMs = connectTimeoutMs;
+	}
+
+	public void setBufferSize(int bufferSize) {
+		Preconditions.checkArgument(connectTimeoutMs > 0, "The buffer size be positive");
```
Comment (uce) [inline flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/PersistentKafkaSource.java]: typo: missing "must"... I think we should add these checks in the constructor as well.

**Comment by uce** [comment]
LGTM but 

```
[INFO] There are 2 checkstyle errors.
[ERROR] KafkaMultiplePartitionsIterator.java[37:n/a] Line has leading space characters; indentation should be performed with tabs only.
[ERROR] KafkaMultiplePartitionsIterator.java[38:n/a] Line has leading space characters; indentation should be performed with tabs only.
```

**Comment by rmetzger** [comment]
Thanks for the review @uce. I'll update the PR.

**Comment by rmetzger** [comment]
I'updated the PR. Once Travis is green, I'll merge it.

**Pull request merged**